sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java

```diff
@@ -53,20 +54,33 @@ public final class SchemasGetByIdHeaders {
     @JsonProperty(value = "Content-Type")
     private String contentType;
 
+    private static final HttpHeaderName SCHEMA_VERSION = HttpHeaderName.fromString("Schema-Version");
+
+    private static final HttpHeaderName SCHEMA_ID = HttpHeaderName.fromString("Schema-Id");
+
+    private static final HttpHeaderName SCHEMA_GROUP_NAME = HttpHeaderName.fromString("Schema-Group-Name");
+
+    private static final HttpHeaderName SCHEMA_NAME = HttpHeaderName.fromString("Schema-Name");
+
+    private static final HttpHeaderName SCHEMA_ID_LOCATION = HttpHeaderName.fromString("Schema-Id-Location");
+
     // HttpHeaders containing the raw property values.
     /**
      * Creates an instance of SchemasGetByIdHeaders class.
      *
      * @param rawHeaders The raw HttpHeaders that will be used to create the property values.
      */
     public SchemasGetByIdHeaders(HttpHeaders rawHeaders) {
-        this.schemaVersion = Integer.parseInt(rawHeaders.getValue("Schema-Version"));
-        this.schemaId = rawHeaders.getValue("Schema-Id");
-        this.schemaGroupName = rawHeaders.getValue("Schema-Group-Name");
-        this.schemaName = rawHeaders.getValue("Schema-Name");
-        this.schemaIdLocation = rawHeaders.getValue("Schema-Id-Location");
-        this.location = rawHeaders.getValue("Location");
-        this.contentType = rawHeaders.getValue("Content-Type");
+        String schemaVersion = rawHeaders.getValue(SCHEMA_VERSION);
+        if (schemaVersion != null) {
+            this.schemaVersion = Integer.parseInt(schemaVersion);
+        }
```

Comment:
Previously, we didn't check for null and if the schema version was null, we'd throw an exception. This will change the behavior.

Reply:
This seems to be changed by Autorest generation.
We can hand revert it with a note here. 
Open to other ideas.